### ISSUE TITLE
Confusing paragraph regarding operator precedence in Exercise 4 readme

### ISSUE THREAD

**Issue by jw013** [tfpk/macrokata#44]
https://github.com/tfpk/macrokata/blob/7d218477eb0abcbf113d1247a23d7e83a0c7310f/exercises/04_expression_variables/README.md?plain=1#L21

This says that if `3 * math!(...)` expands to `3 * 4 + 2`, then the result would evaluate to 14. This hypothetical example confused me because I am not aware of any way to write a macro `math!` such that `3 * math!(4, plus, 2)` expands to `3 * 4 + 2` with a result of 14 instead of `3 * (4 + 2)` with a result of 18. The straight-forward attempt results in 18 despite not using parentheses:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=fc8f2afa0be0dc9f79da7022b02127ed

It seems like in practice every macro invocation is isolated from surrounding code by a virtual pair of parentheses.

**Comment by tfpk** [comment]
Hi!

You're entirely right... I'm trying to figure out where I got this information from. Even `cargo-expand` shows there are brackets:

```rust
macro_rules! math {
    () => {3 + 4}
}

fn main() {
    let _ = 3 * math!();
}
```
expands to...
```rust
fn main() {
    let _ = 3 * (3 + 4);
}
```


I'll update the book. Thank you!!

**Comment by tfpk** [comment]
Fixed in a30fdbe, and I made it clear that a previous version of the book was wrong (and that this isn't clarified anywhere, which is I suppose why I didn't catch this mistake).

Thank you for pointing this out!
